lib/SILOptimizer/LoopTransforms/LICM.cpp

```diff
@@ -216,14 +227,20 @@ static bool sinkInstruction(DominanceInfo *DT,
 
   bool Changed = false;
   for (auto *ExitingBB : ExitingBBs) {
+    SmallVector<SILBasicBlock *, 8> BBSuccessors;
```

Comment:
Can you hoist this SmallVector our of the loop?

Reply:
To be clear, I am talking about hoisting it out of the loop and then clearing before each loop iteration.

Reply:
One exiting bb covers most loops, in fact we didn't bother support multiple basic blocks before the refactoring, loop will execute 1-2 iterations in the vast majority of cases, thought this is shorter / more readable.